### PR TITLE
fix zeromq subscribe proplem

### PR DESCRIPTION
**Pull request by jerry-yu**
* fix read env var to avoid conflict
* use lazy static
fmt

### CODE REVIEW

File: pubsub_zeromq/src/lib.rs [hunk 1]
```diff
@@ -17,122 +17,370 @@
...
+            assert!(
+                publisher
+                    .bind(&*format!(
+                        "{}://{}{}{}",
+                        con_type,
+                        wild_cast,
+                        delimiter,
+                        base_port + idx
+                    ))
+                    .is_ok()
```
Comment (u2) [inline pubsub_zeromq/src/lib.rs]:
```
publisher.bind().expect("")
```

File: pubsub_zeromq/src/lib.rs [hunk 2]
```diff
@@ -17,122 +17,370 @@
...
 
-    let _ = thread::Builder::new()
-        .name("publisher".to_string())
-        .spawn(move || loop {
-            let ret = rx.recv();
+    /*URL like tcp://8.8.8.8:6000  or like ipc://ipc6000*/
+    let _ = SERVICE_PORT_INDEX.get(&name).map_or_else(
+        || {
+            publish_flag = false;
+            info!("synchronizer not need publish !");
```
Comment (u2) [inline pubsub_zeromq/src/lib.rs]: What's the meaning of this info?

File: pubsub_zeromq/src/lib.rs [hunk 3]
```diff
@@ -17,122 +17,370 @@
...
+                publisher.send(&msg, 0).unwrap();
+            });
+    }
 
     //sub
 
-    let network_subscriber = context.socket(zmq::SUB).unwrap();
-    assert!(network_subscriber.connect("tcp://localhost:5563").is_ok());
+    let network_subscriber = CONTEXT.socket(zmq::SUB).unwrap();
+    assert!(
```
Comment (u2) [inline pubsub_zeromq/src/lib.rs]: Should not use `assert!` here, the same as others.

File: pubsub_zeromq/src/lib.rs [hunk 4]
```diff
@@ -17,122 +17,370 @@
...
+        net_tx_subscriber
+            .connect(&*format!(
+                "{}://{}{}{}",
+                con_type,
+                ip,
+                delimiter,
+                base_port + 7
+            ))
+            .is_ok()
+    );
```
Comment (u2) [inline pubsub_zeromq/src/lib.rs:152]: There are so many duplicated codes.

**Review by u2 — CHANGES_REQUESTED** [review]

File: pubsub_zeromq/src/lib.rs (hunk 1 above)
Comment (jerry-yu) [reply pubsub_zeromq/src/lib.rs]: all fixed

**Comment by u2** [comment]
Close it firstly, reopen it later if we really need it.

**Comment by kaikai1024** [comment]
please fix the comment and conflicts. @jerry-yu

**Comment by kaikai1024** [comment]
> please fix the comment and conflicts. @jerry-yu

you can reopen it when you have time.

**Pull request closed without merging**